fpx/src/data.rs

```diff
@@ -228,37 +227,90 @@ impl Store {
         tx: &Transaction<'_, T>,
         method: &str,
         url: &str,
-        body: &str,
+        body: Option<&str>,
         headers: BTreeMap<String, String>,
-    ) -> Result<u32> {
+    ) -> Result<models::Request> {
         let headers = serde_json::to_string(&headers)?;
 
         let request: models::Request = tx
             .query(
                 "INSERT INTO requests (method, url, body, headers) VALUES (?, ?, ?, ?) RETURNING *",
-                (method, url, body, headers),
+                params!(method, url, body, headers),
             )
             .await
             .context("Unable to create request")?
             .fetch_one()
             .await?;
 
-        Ok(request.id)
+        Ok(request)
+    }
+
+    pub async fn request_list<T: ReadTransaction>(
+        &self,
+        tx: &Transaction<'_, T>,
+    ) -> Result<Vec<models::Request>> {
```

Comment:
We should make the model more public, probably can just add `pub` to the struct.